### PR TITLE
correction for sar score normalization

### PR DESCRIPTION
**Pull request by gramhagen**
### Description
update to sar single node normalization scheme

using unity-user-affinity dot item-similarity to generate total counts for each user-item combination during scoring
per user we take the min * min rating value, and max time max rating to rescale the scores of the sar output
this will provide a normalization approach which maintains the same per-user ranking as the un-normalized results, but keeps them within the original rating value range

also removing rescaling and updating tests

### CODE REVIEW

**Comment by gramhagen** [comment]
oops, need to update the rating scores in the readme

**Comment by gramhagen** [comment]
k, fixed

**Comment by viktorku** [comment]
@gramhagen good stuff! Thanks for this. I'll check it out in more detail tomorrow or so, and test it on a few datasets, but seems pretty good and logical 👍

File: reco_utils/recommender/sar/sar_singlenode.py [hunk 1]
```diff
@@ -319,7 +321,11 @@ def score(self, test, remove_seen=False):
             test_scores = test_scores.toarray()
 
         if self.normalize:
-            test_scores = rescale(test_scores, self.rating_scale_min, self.rating_scale_max, data_min=0)
+            counts = self.unity_user_affinity[user_ids, :].dot(self.item_similarity)
+            user_min_scores = np.tile(counts.min(axis=1)[:, np.newaxis], test_scores.shape[1]) * self.rating_min
+            user_max_scores = np.tile(counts.max(axis=1)[:, np.newaxis], test_scores.shape[1]) * self.rating_max
+            test_scores = (test_scores - user_min_scores) / (user_max_scores - user_min_scores)
+            test_scores = test_scores * (self.rating_max - self.rating_min) + self.rating_min
```
Comment (viktorku) [inline reco_utils/recommender/sar/sar_singlenode.py]: This is actually identical to the `rescale` implementation I added. Can we keep that one and pass the per-user min/max scores of the "old" scale?

In that case `rescale` doesn't need to change at all, but would be called as such:

```py
counts = self.unity_user_affinity[user_ids, :].dot(self.item_similarity)
user_min_scores = np.tile(counts.min(axis=1)[:, np.newaxis], test_scores.shape[1]) * self.rating_min
user_max_scores = np.tile(counts.max(axis=1)[:, np.newaxis], test_scores.shape[1]) * self.rating_max
test_scores = rescale(test_scores, self.rating_min, self.rating_max, user_min_scores, user_max_scores)
```

Comment (gramhagen) [reply reco_utils/recommender/sar/sar_singlenode.py]: Yeah the makes sense

**Review by viktorku — APPROVED** [review]
I still haven't run it on some bigger datasets, will do so in a few days, but LGTM

**Pull request merged**